core/src/main/java/org/apache/calcite/sql/SqlDialect.java

```diff
@@ -448,6 +448,15 @@ public void unparseCall(SqlWriter writer, SqlCall call, int leftPrec,
       int rightPrec) {
     SqlOperator operator = call.getOperator();
     switch (call.getKind()) {
+    case JOIN:
+      SqlJoin join = (SqlJoin) call;
+      JoinRelType joinRelType = JoinRelType.convertToRelType(join.getJoinType());
```

Comment:
Maybe you can use [convertJoinType](https://github.com/apache/calcite/blob/12e7d621bbb19aa32cb45329cb84532115037b7a/core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java#L3480)